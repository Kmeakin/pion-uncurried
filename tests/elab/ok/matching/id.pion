enum Id(A: Type) {
    MkId(_: A),
}

let get = fn(A: Type, it: Id(A)) => match it {
    MkId(x) => x,
};