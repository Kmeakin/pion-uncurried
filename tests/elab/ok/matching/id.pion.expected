enum Id(A: Type): Type {
    MkId(_: A): Id(A),
}

let get: fn(A: Type, it: Id(A)) -> A = fn(A: Type, it: Id(A)) => match it {
    MkId(x) => x,
};
